### ISSUE TITLE
Cannot extend configuration files recursively ("extends" with presets)

### ISSUE THREAD

**Issue by christophercr** [palantir/tslint#3874]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.6.2
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

Configuration hierarchy using presets ([tslint-sonarts](https://github.com/SonarSource/SonarTS), [tslint-config-prettier](https://github.com/alexjoverm/tslint-config-prettier), [codelyzer](https://github.com/mgechev/codelyzer)) in the base configuration file:

**tslint-base.json:**
```json
{
  "extends": [
    "tslint:latest",
    "tslint-sonarts",
    "codelyzer",
    "tslint-config-prettier"
  ],
  "defaultSeverity": "error",
  "rules": { ... }
}
```

**tslint.json:**
```json
{
  "extends": ["./path/to/tslint-base.json"],
  "defaultSeverity": "warning",
  "rules": { ... }
}
```

#### Actual behavior
The presets included in the "extends" option of the `tslint-base.json` are not taken so the linting does not report any violation of those rules.
In order to make those rules work, the presets should be moved to the child `tslint.json`:

**tslint-base.json:**
```json
{
  // removed the "extends" option from here
  "defaultSeverity": "error",
  "rules": { ... }
}
```

**tslint.json:**
```json
{
  "extends": [
    "tslint:latest",
    "tslint-sonarts",
    "codelyzer",
    "tslint-config-prettier"
    "./path/to/tslint-base.json"
  ],
  "defaultSeverity": "warning",
  "rules": { ... }
}
```

Although this post show explains how to create [sharable configurations and rules](https://palantir.github.io/tslint/2016/03/31/sharable-configurations-rules.html), it doesn't cover the use case mentioned in this issue (with recursive "extends" and using presets). Basically the same functionality provided by [ESLint: extending configuration files](https://eslint.org/docs/user-guide/configuring#extending-configuration-files)

#### Expected behavior
The presets should be correctly taken in the base tslint.json and be overwriten by any rules, extensions used in the child tslint.json.

**Comment by giladgray** [comment]
@christophercr i'm pretty sure this works... we rely on it every day.
